content/en/localstack/init-hooks.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "Initialization hooks"
+weight: 5
+description: >
+  Write shell or Python scripts to customize or initialize your LocalStack instance.
+---
+
+## Lifecycle stages and hooks
+
+LocalStack has four well-known lifecycle phases or stages:
+* `BOOT`: the container is running but localstack runtime has not been started
+* `START`: the Python process is running and the localstack runtime is starting
+* `READY`: localstack is ready to serve requests
+* `SHUTDOWN`: localstack is shutting down
+
+You can hook into each of these lifecycle phases using custom shell or python scripts.
+Each lifecycle phase has it's own directory in `/etc/localstack/init`.
+You can mount individual files, stage directories, or the entire init directory from your host into the container.
+
+```
+/etc
+└── localstack
+    └── init
+        ├── boot.d           <-- executed in the container before localstack starts
+        ├── ready.d          <-- executed when localstack becomes ready (currently equivalent to `/docker-entrypoint-initaws.d`)
+        ├── shutdown.d       <-- executed when localstack shuts down
+        └── start.d          <-- executed when localstack starts up
+```
+
+In these directories, you can put either executable shell scripts or python programs, which will be executed from within a Python process.
+All except `boot.d` will be run in the same Python interpreter as LocalStack, which gives additional ways of configuring/extending LocalStack.
+
+Currently, known script extensions are `.sh`, and `.py`.
+Shell scripts have to be executable.
+
+A script can be in one of four states: `UNKNOWN`, `RUNNING`, `SUCCESSFUL`, `ERROR`.
+Scripts are by default in the `UNKNOWN` state once they have been discovered.
+The remaining states should be self-explanatory.
+
+### Execution order and script failures
+
+Scripts are sorted and executed in alphanumerical order.
+If an init script fails, the remaining scripts will still be executed in order.
+A script is considered in `ERROR` state if it is a shell script and returns with a non-zero exit code, or if a Python script raises an exception during its execution.
+
+
+## Status endpoint
+
+There is an additional endpoint in `localhost:4566/_localstack/init` which returns the state of the initialization procedure.
+Boot scripts are currently always in the unknown state, since they are executed outside the localstack process and we don't know whether they have been successfully executed or not.
```

Comment:
```suggestion
Boot scripts (scripts placed in `boot.d`) are currently always in the `UNKNOWN` state, since they are executed outside the LocalStack process and we don't know whether they have been successfully executed or not.
```